error: Definition of "_meta" in base class "PermissionsMixin" is incompatible with definition in base class "AbstractBaseUser"

# Bug report

<!--
Hi, thanks for submitting a bug. We appreciate that.

But, we will need some information about what's wrong to help you.
-->

## What's wrong

We have defined a custom `AuthUser` model in our project like:

```python
from django.contrib.auth.models import AbstractBaseUser, PermissionsMixin

class AuthUser(PermissionsMixin, AbstractBaseUser):
    class Meta:
        verbose_name = _('Auth User')
        abstract = False
        db_table = 'auth_user'
```

mypy gives an error:
```
myproject/models/auth.py:3: error: Definition of "_meta" in base class "PermissionsMixin" is incompatible with definition in base class "AbstractBaseUser"
```
(from the `class AuthUser(PermissionsMixin, AbstractBaseUser):` line)

The slightly weird part is Django's `PermissionMixin` and `AbstractBaseUser` both have an identical `Meta`:
```python
    class Meta:
        abstract = True
```

## How is that should be

No error

## System information

- OS:
- `python` version: 3.8.3
- `django` version: 2.2.12
- `mypy` version: 0.782
- `django-stubs` version: 1.5.0

```
myproject/models/auth.py:19: error: Definition of "_meta" in base class "PermissionsMixin" is incompatible with definition in base class "AbstractBaseUser"
myproject/models/auth.py:111: note: Revealed type is 'def (*args: Any, **kwargs: Any) -> myproject.models.auth.AuthUser'
```

Reproduced:
```
git clone https://github.com/syastrov/django-stubs-471.git
cd django-stubs-471
python3 -m venv venv
source ./venv/bin/activate
pip3 install -r requirements.txt 
mypy app1/
```

Output
```
authtest/settings.py:28: error: Need type annotation for 'ALLOWED_HOSTS' (hint: "ALLOWED_HOSTS: List[<type>] = ...")
app1/models.py:4: error: Definition of "_meta" in base class "PermissionsMixin" is incompatible with definition in base class "AbstractBaseUser"
Found 2 errors in 2 files (checked 7 source files)
```

Disabling the plugin removes the error...

--- a/django-stubs/db/models/base.pyi
+++ b/django-stubs/db/models/base.pyi
@@ -41,7 +41,7 @@ class Model(metaclass=ModelBase):
     objects: ClassVar[Manager[Self]]
 
     class Meta: ...
-    _meta: Options[Any]
+    _meta: ClassVar[Options[Self]]
     pk: Any
     _state: ModelState
     def __init__(self, *args: Any, **kwargs: Any) -> None: ...

--- a/mypy_django_plugin/transformers/models.py
+++ b/mypy_django_plugin/transformers/models.py
@@ -613,13 +613,6 @@ class AddExtraFieldMethods(ModelClassInitializer):
                 )
 
 
-class AddMetaOptionsAttribute(ModelClassInitializer):
-    def run_with_model_cls(self, model_cls: Type[Model]) -> None:
-        if "_meta" not in self.model_classdef.info.names:
-            options_info = self.lookup_typeinfo_or_incomplete_defn_error(fullnames.OPTIONS_CLASS_FULLNAME)
-            self.add_new_node_to_model_class("_meta", Instance(options_info, [Instance(self.model_classdef.info, [])]))
-
-
 class ProcessManyToManyFields(ModelClassInitializer):
     """
     Processes 'ManyToManyField()' fields and generates any implicit through tables that
@@ -918,7 +911,6 @@ def process_model_class(ctx: ClassDefContext, django_context: DjangoContext) -> 
         AddDefaultManagerAttribute,
         AddReverseLookups,
         AddExtraFieldMethods,
-        AddMetaOptionsAttribute,
         ProcessManyToManyFields,
         MetaclassAdjustments,
     ]

--- a/tests/typecheck/models/test_contrib_models.yml
+++ b/tests/typecheck/models/test_contrib_models.yml
@@ -62,3 +62,32 @@
                   ...
               class MyUser(AbstractUser):
                   objects: ClassVar[MyUserManager] = MyUserManager()
+
+-   case: can_combine_permissions_mixin_and_abstract_base_user
+    main: |
+        from django.contrib.auth.base_user import AbstractBaseUser
+        from django.contrib.auth.models import PermissionsMixin
+        from myapp.models import AuthUser
+        reveal_type(AuthUser._meta)
+        reveal_type(AuthUser()._meta)
+        reveal_type(PermissionsMixin._meta)
+        reveal_type(AbstractBaseUser._meta)
+    out: |
+        main:4: note: Revealed type is "django.db.models.options.Options[myapp.models.AuthUser]"
+        main:5: note: Revealed type is "django.db.models.options.Options[myapp.models.AuthUser]"
+        main:6: note: Revealed type is "django.db.models.options.Options[django.contrib.auth.models.PermissionsMixin]"
+        main:7: note: Revealed type is "django.db.models.options.Options[django.contrib.auth.base_user.AbstractBaseUser]"
+    installed_apps:
+        - django.contrib.auth
+        - myapp
+    files:
+        - path: myapp/__init__.py
+        - path: myapp/models.py
+          content: |
+              from django.contrib.auth.base_user import AbstractBaseUser
+              from django.contrib.auth.models import PermissionsMixin
+
+              class AuthUser(AbstractBaseUser, PermissionsMixin):
+                  class Meta:
+                      abstract = False
+                      db_table = "auth_user"

--- a/tests/typecheck/models/test_meta_options.yml
+++ b/tests/typecheck/models/test_meta_options.yml
@@ -40,7 +40,7 @@
         from myapp.models import AbstractModel
         class MyModel(AbstractModel):
             pass
-        reveal_type(MyModel._meta.get_field('field'))  # N: Revealed type is "Any"
+        reveal_type(MyModel._meta.get_field('field'))  # N: Revealed type is "Union[django.db.models.fields.Field[Any, Any], django.db.models.fields.reverse_related.ForeignObjectRel, django.contrib.contenttypes.fields.GenericForeignKey]"
     installed_apps:
         - myapp
     files:

